src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -225,6 +226,29 @@ public override void OnCustom(string name, Activity activity, object payload)
             }
         }
 
+        public override void OnException(Activity activity, object payload)
+        {
+            if (activity.IsAllDataRequested && this.options.RecordException)
+            {
+                if (!this.stopExceptionFetcher.TryFetch(payload, out Exception exc) || exc == null)
+                {
+                    AspNetCoreInstrumentationEventSource.Log.NullPayload(nameof(HttpInListener), nameof(this.OnException));
+                    return;
+                }
+
+                try
+                {
+                    this.options.Enrich?.Invoke(activity, "OnException", exc);
+                }
+                catch (Exception ex)
+                {
+                    AspNetCoreInstrumentationEventSource.Log.EnrichmentException(ex);
+                }
+
+                activity.SetStatus(Status.Error.WithDescription(exc.Message));
```

Comment:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/exceptions.md Lets populate this into ActivityEvent.

Reply:
just updated and added in another place that was missing